kuksa_databroker/databroker/src/grpc/kuksa_val_v1/val.rs

```diff
@@ -54,55 +55,156 @@ impl proto::val_server::Val for broker::DataBroker {
+                                                                .to_owned(),
+                                                        }),
+                                                    });
+                                                }
+                                            }
+                                            Err(ReadError::NotFound) => {
+                                                errors.push(proto::DataEntryError {
+                                                    path: request.path,
+                                                    error: Some(proto::Error {
+                                                        code: 404,
+                                                        reason: "not_found".to_owned(),
+                                                        message:
+                                                            "No entries found for the provided path"
+                                                                .to_owned(),
+                                                    }),
+                                                });
+                                            }
+                                            Err(ReadError::PermissionExpired) => {
+                                                errors.push(proto::DataEntryError {
+                                                    path: request.path,
+                                                    error: Some(proto::Error {
+                                                        code: 401,
+                                                        reason: "unauthorized".to_owned(),
+                                                        message: "Authorization expired".to_owned(),
+                                                    }),
+                                                });
+                                            }
+                                            Err(ReadError::PermissionDenied) => {
+                                                errors.push(proto::DataEntryError {
+                                                    path: request.path,
+                                                    error: Some(proto::Error {
+                                                        code: 403,
+                                                        reason: "forbidden".to_owned(),
+                                                        message: "Permission denied".to_owned(),
+                                                    }),
+                                                });
+                                            }
+                                        }
+                                    }
+                                    Err(_error) => {
+                                        println!("Error: Invalid Regex");
+                                    }
```

Comment:
Return error to client?

Reply:
You are right, should not return anything